### DISCUSSION TITLE
Contract Initialization Error on Testnet

### DISCUSSION THREAD

**Discussion by knfa28** [TENK-DAO/tenk#103] [Q&A] [answered]
Hello! I am trying to deploy and initialize the TenK NFT template contract (https://github.com/TENK-DAO/tenk/tree/main/contracts/tenk) over at testnet, however after calling this command on the terminal:
```
near call 123.xyz.testnet new '{"owner_id": "123.xyz.testnet", "metadata": {"name": "Test TenK NFT Contract", "symbol": "TEST", "spec": "NEP171"}, "sale": {"price": "69"}, "size": 420}' --accountId 123.xyz.testnet
```
I am getting the following error:

![error](https://user-images.githubusercontent.com/10632537/176334583-3ecbb5ef-6aaf-4303-82ee-1aa4a79abcbc.PNG)

Any advice on how should I proceed?

**Comment by willemneal** [comment, marked as answer]
I would recommend using the other init method https://github.com/TENK-DAO/tenk/blob/ac7831f7b0547bb84edf2de1407908ae440abdff/contracts/tenk/src/lib.rs#L94-L100

As it sets the default for the spec, which you set incorrectly, see [here](https://github.com/near/near-sdk-rs/blob/6d4045251c63ec875dc55f43b065b33a36d94792/near-contract-standards/src/non_fungible_token/metadata.rs#L7)
